ui/src/app/cron-workflows/cron-workflow-details.tsx

```diff
@@ -36,9 +37,16 @@ export function CronWorkflowDetails({match, location, history}: RouteComponentPr
     const [columns, setColumns] = useState<models.Column[]>([]);
 
     const [cronWorkflow, setCronWorkflow] = useState<CronWorkflow>();
-    const [edited, setEdited] = useState(false);
+    const [initialCronWorkflow, setInitialCronWorkflow] = useState<CronWorkflow>();
```

Comment:
Nice analysis ! You could use a Ref for `initialCronWorkflow` as you don't expect a rerendering in your case.

Also you could implement a more reusable logic in a separate testable hook as you reuse it in multiple places

I was thinking about something like (not tested) : 

```
const useStateIsEdited = function <T>(initialStateValue?: T) {
  const [initialValue, setInitialValue] = useState<T | undefined>(initialStateValue)
  const [value, setValue] = useState<T>()

  const edited = useMemo(() => !isEqual(value, initialValue), [value, initialValue])

  return [value, setValue, setInitialValue, edited] as const
}
```

That we reuse everywhere by calling
```
const [cronWorkflow, setCronWorkflow, setInitialCronWorkflow,, edited] = useStateIsEdited<T>()
```

Otherwise LGTM

Reply:
Yea I was thinking we could refactor a hook here too. It's a pretty time one, but perhaps worth the encapsulation. I don't really have a strong opinion on it

Reply:
@Adrien-D That's a good idea.  I pushed 24c0206b9071503467034cb67716ef0b27158b0f to create a `useEditableResource()` hook. I didn't write tests since I think we'd need to add [@testing-library/react](https://github.com/testing-library/react-testing-library) as a dependency first.

Reply:
Yea I'd like to add tests with `testing-library` eventually to the UI as it is sorely lacking them and some places have fairly complex logic